sherpa/plot/backends.py

```diff
@@ -0,0 +1,961 @@
+#
+#  Copyright (C) 2007, 2015, 2020, 2021, 2022, 2023
+#  Smithsonian Astrophysical Observatory
+#
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License along
+#  with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+from copy import deepcopy
+from inspect import signature
+import logging
+import types
+
+from sherpa.utils import formatting
+from sherpa.plot.backend_utils import (translate_args,
+                                       add_kwargs_to_doc,
+                                       get_keyword_defaults)
+
+__all__ = ('BasicBackend',
+           'backend_indep_kwargs',
+           'IndepOnlyBackend',
+           'PLOT_BACKENDS',
+           )
+
+PLOT_BACKENDS = {}
+'''global list of all successfully imported plotting backends'''
+
+lgr = logging.getLogger(__name__)
+warning = lgr.warning
+
+backend_indep_colors = list('rgbkwcymk') + [None]
```

Comment:
There are a number of 'Pending' comments to this file in #1382 that apply to this file. For example, "rgb**k**wcym**k**" lists black twice.

Reply:
Then there is no need to comment on them here; I plan to rebase this on main after #1382 is merged.

Reply:
@wmclaugh I don't see those comments. If they are "pending" on your account, you have not submitted them yet. There should be a "finish review" button or something like that that will actually post all the comments that you wrote so that I can see them.

Reply:
@hamogu-  Apologies- you are correct. They should be visible now.